rest/perf_routes.go

```diff
@@ -358,7 +358,70 @@ func (h *perfSignalProcessingRecalculateHandler) Run(ctx context.Context) gimlet
 		grip.Error(message.WrapError(err, message.Fields{
 			"request": gimlet.GetRequestID(ctx),
 			"method":  "POST",
-			"route":   "/perf/signal_processing/recalculate",
+			"route":   "/perf/change_points",
+		}))
+		return gimlet.MakeJSONErrorResponder(err)
+	}
+	return gimlet.NewJSONResponse(struct{}{})
+}
+
+///////////////////////////////////////////////////////////////////////////////
+//
+// POST /perf/change_points/triage/mark
+
+type changePointStub struct {
+	PerfResultID string `bson:"perf_result_id" json:"perf_result_id" yaml:"perf_result_id"`
+	Measurement  string `bson:"measurement" json:"measurement" yaml:"measurement"`
+}
+
+type changePointMarkRequest struct {
+	Status       string            `bson:"status" json:"status" yaml:"status"`
+	ChangePoints []changePointStub `bson:"change_points" json:"change_points" yaml:"change_points"`
+}
+
+type perfChangePointTriageMarkHandler struct {
+	sc  data.Connector
+	req changePointMarkRequest
+}
+
+func makePerfChangePointTriageMarkHandler(sc data.Connector) gimlet.RouteHandler {
+	return &perfChangePointTriageMarkHandler{
+		sc: sc,
+	}
+}
+
+func (h *perfChangePointTriageMarkHandler) Factory() gimlet.RouteHandler {
+	return &perfChangePointTriageMarkHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *perfChangePointTriageMarkHandler) Parse(ctx context.Context, r *http.Request) error {
+	err := gimlet.GetJSON(r.Body, h.req)
+	if err != nil {
+		err = errors.Wrapf(err, "Error parsing triage status change request")
+		grip.Error(message.WrapError(err, message.Fields{
+			"request": gimlet.GetRequestID(ctx),
+			"method":  "POST",
+			"route":   "/perf/change_points/triage/mark",
+		}))
+		return err
+	}
+	return nil
+}
+
+func (h *perfChangePointTriageMarkHandler) Run(ctx context.Context) gimlet.Responder {
+	cps := map[string]string{}
+	for _, stub := range h.req.ChangePoints {
+		cps[stub.PerfResultID] = stub.Measurement
+	}
+	err := h.sc.TriageChangePoints(ctx, cps, h.req.Status)
```

Comment:
I'd be inclined to pass the slice rather than a map around.

Reply:
So you mentioned this earlier:
* I don't want to pass a slice of structs in this package down to model package (circular dep)
* I had issues with passing slices of an anonymous struct

Was I just doing point 2 the wrong way?

Reply:
Update above.